### PR TITLE
add entry about diffusers

### PR DESCRIPTION
**Pull request by sayakpaul**

### CODE REVIEW

File: docs/hub/diffusers.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Using 🧨 `diffusers` at Hugging Face
+
+🧨 `diffusers` is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. Whether you’re looking for a simple inference solution or want to train your own diffusion model, 🤗 Diffusers is a modular toolbox that supports both. Our library is designed with a focus on usability over performance, simple over easy, and customizability over abstractions.
```
Comment (osanseviero) [inline docs/hub/diffusers.md:3]:
```suggestion
🧨 `diffusers` is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. Whether you’re looking for a simple inference solution or want to train your own diffusion model, 🤗 Diffusers is a modular toolbox that supports both. The library is designed with a focus on usability over performance, simple over easy, and customizability over abstractions.
```

File: docs/hub/diffusers.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Exploring 🧨 diffusers in the Hub
+
+There are over 10,000 `diffusers` compatible pipelines on the Hub which you can find by filtering at the left of [the models page](https://huggingface.co/models?library=diffusers&sort=downloads). Diffusion systems are typically composed of multiple components such as text encoder, UNet, VAE, and scheduler. This is why they are not standalone models. 
+
+You can find diffusion pipelines for many different tasks:
+
+* Generating images from natural language text prompt ([text-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=text-to-image&sort=downloads)).
+* Translating images from images using natural language text prompt ([image-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=image-to-image&sort=downloads)).
+* Generating videos from natural language text prompt ([text-to-video](https://huggingface.co/models?library=diffusers&pipeline_tag=text-to-video&sort=downloads)).
```
Comment (osanseviero) [inline docs/hub/diffusers.md:13]: WDYT about this minor rewording?

```suggestion
* Generating images from natural language text prompts ([text-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=text-to-image&sort=downloads)).
* Creating new images based on input images and text prompts ([image-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=image-to-image&sort=downloads)).
* Generating videos from text prompts ([text-to-video](https://huggingface.co/models?library=diffusers&pipeline_tag=text-to-video&sort=downloads)).
```

File: docs/hub/diffusers.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+
+You can try out the models directly in the browser if you want to test them out without downloading them thanks to the in-browser widgets!
```
Comment (osanseviero) [inline docs/hub/diffusers.md:16]:
```suggestion
You can try out the models directly in the browser if you want to test them out without downloading them, thanks to the in-browser widgets! 
```

**Review by osanseviero — APPROVED** [review]
Very nice!! 🔥

**Review by Wauplin — COMMENTED** [review]
Thanks for adding this @sayakpaul! Looks good to me. Could you also add `diffusers` to the table [on this page](https://huggingface.co/docs/hub/models-libraries)? It has to be manually edited in https://github.com/huggingface/hub-docs/blob/main/docs/hub/models-libraries.md.

File: docs/hub/_toctree.yml [hunk 4]
```diff
@@ -57,6 +57,8 @@
     sections:
       - local: transformers
         title: 🤗 transformers
+      - local: diffusers
+        title: 🧨 diffusers
```
Comment (julien-c) [inline docs/hub/_toctree.yml:61]: not sure about the emoji here, i would remove it

```suggestion
        title: Diffusers
```

Comment (julien-c) [reply docs/hub/_toctree.yml:61]: (i would also remove it from Transformers in another PR, tbh)

Also can you keep the alphabetical order?

File: docs/hub/diffusers.md (hunk 1 above)
Comment (julien-c) [reply docs/hub/diffusers.md:3]: i would remove (most of) the emojis here too

File: docs/hub/_toctree.yml (hunk 4 above)
Comment (osanseviero) [reply docs/hub/_toctree.yml:61]: (it was alphabetical with the emojis hehe)

File: docs/hub/diffusers.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Using 🧨 `diffusers` at Hugging Face
+
+🧨 `diffusers` is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. Whether you’re looking for a simple inference solution or want to train your own diffusion model, 🤗 Diffusers is a modular toolbox that supports both. Our library is designed with a focus on usability over performance, simple over easy, and customizability over abstractions.
+
+## Exploring 🧨 diffusers in the Hub
+
+There are over 10,000 `diffusers` compatible pipelines on the Hub which you can find by filtering at the left of [the models page](https://huggingface.co/models?library=diffusers&sort=downloads). Diffusion systems are typically composed of multiple components such as text encoder, UNet, VAE, and scheduler. This is why they are not standalone models.
```
Comment (pcuenca) [inline docs/hub/diffusers.md:7]: thoughts about using `trending` sorting instead?

Comment (pcuenca) [inline docs/hub/diffusers.md:7]:
```suggestion
There are over 10,000 `diffusers` compatible pipelines on the Hub which you can find by filtering at the left of [the models page](https://huggingface.co/models?library=diffusers&sort=downloads). Diffusion systems are typically composed of multiple components such as text encoder, UNet, VAE, and scheduler. Even though they are not standalone models, the pipeline abstraction makes it easy to use them for inference or training.
```

File: docs/hub/diffusers.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+🧨 `diffusers` is the go-to library for state-of-the-art pretrained diffusion models for generating images, audio, and even 3D structures of molecules. Whether you’re looking for a simple inference solution or want to train your own diffusion model, 🤗 Diffusers is a modular toolbox that supports both. Our library is designed with a focus on usability over performance, simple over easy, and customizability over abstractions.
+
+## Exploring 🧨 diffusers in the Hub
+
+There are over 10,000 `diffusers` compatible pipelines on the Hub which you can find by filtering at the left of [the models page](https://huggingface.co/models?library=diffusers&sort=downloads). Diffusion systems are typically composed of multiple components such as text encoder, UNet, VAE, and scheduler. This is why they are not standalone models. 
+
+You can find diffusion pipelines for many different tasks:
+
+* Generating images from natural language text prompt ([text-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=text-to-image&sort=downloads)).
```
Comment (pcuenca) [inline docs/hub/diffusers.md:11]:
```suggestion
* Generating images from natural language text prompts ([text-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=text-to-image&sort=downloads)).
```

File: docs/hub/diffusers.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+* Translating images from images using natural language text prompt ([image-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=image-to-image&sort=downloads)).
```
Comment (pcuenca) [inline docs/hub/diffusers.md:12]:
```suggestion
* Transforming images using natural language text prompts ([image-to-image](https://huggingface.co/models?library=diffusers&pipeline_tag=image-to-image&sort=downloads)).
```

File: docs/hub/diffusers.md (hunk 2 above)
Comment (pcuenca) [inline docs/hub/diffusers.md:13]:
```suggestion
* Generating videos from natural language descriptions ([text-to-video](https://huggingface.co/models?library=diffusers&pipeline_tag=text-to-video&sort=downloads)).
```

Comment (pcuenca) [reply docs/hub/diffusers.md:13]: I wrote my suggestions before reading Omar's, his works for me too.

File: docs/hub/diffusers.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+You can try out the models directly in the browser if you want to test them out without downloading them thanks to the in-browser widgets! 
+
+<div class="flex justify-center">
+<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/refs%2Fpr%2F35/hub/libraries-diffusers_widget.png"/>
+</div>
+
+## Using existing pipelines
+
+All `diffusers` pipelines are a line away from being used! If you want to run the entire process of generation then it's recommended to always start from the `DiffusionPipeline`:
```
Comment (pcuenca) [inline docs/hub/diffusers.md:24]:
```suggestion
All `diffusers` pipelines are a line away from being used! To run generation we recommended to always start from the `DiffusionPipeline`: 
```

File: docs/hub/diffusers.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+```py
+from diffusers import UNet2DConditionModel
+
+unet = UNet2DConditionModel.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", subfolder="unet")
+```
+
+## Sharing your pipelines and models
+
+All the [pipeline classes](https://huggingface.co/docs/diffusers/main/en/api/pipelines/overview), [model classes](https://huggingface.co/docs/diffusers/main/en/api/models/overview), and [scheduler classes](https://huggingface.co/docs/diffusers/main/en/api/schedulers/overview) are fully compatible with the Hub. More specifically, they can be easily loaded from the Hub using the `from_pretrained()` method and can be shared with others using the `push_to_hub()` method.
```
Comment (pcuenca) [inline docs/hub/diffusers.md:42]:
```suggestion
All the [pipeline classes](https://huggingface.co/docs/diffusers/main/api/pipelines/overview), [model classes](https://huggingface.co/docs/diffusers/main/api/models/overview), and [scheduler classes](https://huggingface.co/docs/diffusers/main/api/schedulers/overview) are fully compatible with the Hub. More specifically, they can be easily loaded from the Hub using the `from_pretrained()` method and can be shared with others using the `push_to_hub()` method.
```

Not sure if we automatically redirect based on browser settings, but we could :)

**Review by pcuenca — APPROVED** [review]

File: docs/hub/diffusers.md (hunk 9 above)
Comment (sayakpaul) [reply docs/hub/diffusers.md:42]: WDYM?

**Comment by julien-c** [comment]
BTW, in https://huggingface.co/docs/hub/models-libraries shouldn't Inference API and widget now be marked as ✅  implemented?

**Comment by pcuenca** [comment]
Yes, I think so! Just opened https://github.com/huggingface/hub-docs/pull/1013 so others can confirm

**Pull request merged**